Week6/Day2/Blahgger/Blahgger/Controllers/AccountController.cs

```diff
@@ -0,0 +1,146 @@
+                    connection.Open();
+                    string sql = @"
+                      SELECT Id, Email, HashedPassword
+                      FROM [User]
+                      Where Email = @Email
+                    ";
+                    SqlCommand command = new SqlCommand(sql, connection);
+                    command.Parameters.AddWithValue("@Email", user.Email);
+
+                    SqlDataReader reader = command.ExecuteReader();
+                    if (reader.Read())
+                    {
+                        if (!Hashing.ValidatePassword(user.Password, reader.GetString(2)))
+                        {
+                            ModelState.AddModelError("", "Login failed.");
+                        }
+                    }
+                    else
+                    {
+                        ModelState.AddModelError("", "Login failed.");
+                    }
+
+                }
+                // TODO check if the passwords match
+                //if (!(user.Email == "james@smashdev.com" && user.Password == "password"))
+                //{
+                //    ModelState.AddModelError("", "Login failed.");
+                //}
+            }
+
+            if (ModelState.IsValid)
+            {
+                // Login the user.
+                FormsAuthentication.SetAuthCookie(user.Email, false);
+
+                // Send them to the home page.
+                return RedirectToAction("Index", "Home");
+            }
+
+            return View(user);
+        }
+
+        [HttpPost]
+        public ActionResult Logout()
+        {
+            FormsAuthentication.SignOut();
+
+            return RedirectToAction("Login");
+        }
+
+        [AllowAnonymous]
+        public ActionResult Register()
+        {
+            return View();
+        }
+
+        [AllowAnonymous]
+        [HttpPost]
+        public ActionResult Register(User user)
+        {
+            //Check if email already exists
+            using (SqlConnection connection = new SqlConnection(connectionString))
+            {
+                connection.Open();
+                string sql = @"
+                      SELECT Id, Email
+                      FROM [User]
+                      Where Email = @Email
+                    ";
+                SqlCommand command = new SqlCommand(sql, connection);
+                command.Parameters.AddWithValue("@Email", user.Email);
+
+                SqlDataReader reader = command.ExecuteReader();
+                if (reader.Read())
+                {
+                    ModelState.AddModelError("", "Email has already been used");
+                }
+                else
+                {
+                    //Hash password
+                    user.HashedPassword = Hashing.HashPassword(user.Password);
+                    //Add account to database
+                    AddUser(user);
+
+                    //Redirect to login
+                    return RedirectToAction("Login");
```

Comment:
You're doing it very often, but be careful about adding code comments that simply read as the code itself does.